packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-file-beta/gux-form-field-file.tsx

```diff
@@ -0,0 +1,358 @@
+import {
+  Component,
+  State,
+  Element,
+  JSX,
+  Prop,
+  h,
+  Listen,
+  forceUpdate
+} from '@stencil/core';
+
+import { buildI18nForComponent, GetI18nValue } from '../../../../../i18n';
+import { ILocalizedComponentResources } from '../../../../../i18n/fetchResources';
+
+import { OnMutation } from '@utils/decorator/on-mutation';
+import { hasSlot } from '@utils/dom/has-slot';
+import { trackComponent } from '@utils/tracking/usage';
+import {
+  GuxFormFieldError,
+  GuxFormFieldContainer,
+  GuxFormFieldHelp,
+  GuxFormFieldLabel
+} from '../../functional-components/functional-components';
+import { GuxFormFieldIndicatorMark } from '../../gux-form-field.types';
+import { getSlottedInput, validateFormIds } from '../../gux-form-field.service';
+import { preventBrowserValidationStyling } from '@utils/dom/prevent-browser-validation-styling';
+import {
+  onDisabledChange,
+  onRequiredChange
+} from '@utils/dom/on-attribute-change';
+
+import componentResources from './i18n/en.json';
+
+/**
+ * @slot input - Required slot for input tag
+ * @slot label - Required slot for label tag
+ * @slot error - Optional slot for error message
+ * @slot help - Optional slot for help message
+ * @slot label-info - Optional slot for tooltip
+ */
+
+@Component({
+  styleUrl: 'gux-form-field-file.scss',
+  tag: 'gux-form-field-file-beta',
+  shadow: true
+})
+export class GuxFormFieldFileBeta {
+  private getI18nValue: GetI18nValue;
+  private dropContainer: HTMLDivElement;
+  private input: HTMLInputElement;
+  private labelInfo: HTMLGuxLabelInfoBetaElement;
+  private disabledObserver: MutationObserver;
+  private requiredObserver: MutationObserver;
+  private hideLabelInfoTimeout: ReturnType<typeof setTimeout>;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   * Field indicator mark which can show *, (optional) or blank
+   * Defaults to required. When set to required, the component will display * for required fields and blank for optional
+   * When set to optional, the component will display (optional) for optional and blank for required.
+   */
+  @Prop()
+  indicatorMark: GuxFormFieldIndicatorMark = 'required';
+
+  @Prop()
+  dropAndDrag: boolean = false;
```

Comment:
I think the phrase "drag and drop" is more commonly used? So I would rename this "dragAndDrop"

Reply:
No idea why I switched that around in the property name 🙈